buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -545,11 +562,65 @@ static int writeUtf8(AbstractByteBuf buffer, int writerIndex, CharSequence seq,
         return writerIndex - oldWriterIndex;
     }
 
+    /**
+     * Returns max bytes length of UTF8 character sequence of the given length.
+     */
+    public static int utf8MaxBytes(final int seqLength) {
+        return seqLength * MAX_BYTES_PER_CHAR_UTF8;
+    }
+
     /**
      * Returns max bytes length of UTF8 character sequence.
+     * <p>
+     * It behaves like {@link #utf8MaxBytes(int)} applied to {@code seq} {@link CharSequence#length()}.
      */
     public static int utf8MaxBytes(CharSequence seq) {
-        return seq.length() * MAX_BYTES_PER_CHAR_UTF8;
+        return utf8MaxBytes(seq.length());
+    }
+
+    /**
+     * Returns the exact bytes length of a well-formed UTF8 character sequence.
+     * <p>
+     * This method is producing a precise estimation only with well-formed UTF8 character sequences, otherwise
```

Comment:
(Excuse the "drive-by" comments): This wording here is a little confusing. "precise estimation" is mentioned here, but the previous sentence mentions "exact", which seems more correct.

Reply:
> but the previous sentence mentions "exact", which seems more correct.

it is just another way to avoid repeating the same sentence `the exact bytes length`, but I understand that "estimation" could be interpreted as something less precise than "exact"